src/main/flight/autopilot.c

```diff
@@ -53,131 +54,138 @@ static pidCoefficient_t positionPidCoeffs;
 static float altitudeI = 0.0f;
 static float throttleOut = 0.0f;
 
-typedef struct earthFrame_s {
+typedef struct {
     bool isStopping;
-    float distance;
     float previousDistance;
     float previousVelocity;
     float integral;
-    float pidSum;
     pt1Filter_t velocityLpf;
     pt1Filter_t accelerationLpf;
-} earthFrame_t;
+} pidAxis_t;
 
 typedef enum {
-    lat = 0,
-    lon
-} axisEF_t;
+    // axes are for ENU system; it is different from current BF code
```

Comment:
Do you mean that by putting lat first in the enum, which matches the gps llh enum order, and the gps_location_t order, I am not following the current Betaflight convention?  Even in MSP and OSD we go latitude first.
At this point I don't understand why you want to put longitude first.

Reply:
It's all about left-handed and right-handed coordinate system. 

Current state in BS is a mess, usually NEU - that would correspond to lat,lon,alt. But this system is left-handed and almost never used.
It can be changed to NED, which is pretty common, but that would mean that altitude will be negative and decrease with increasing height (Z axis pointing down). I think  almost nobody will approve that in BF. And another effect would be Z axis pointing down in body frame, not matching accelerometer etc .. 

> At this point I don't understand why it's important to put longitude first.

If you want to work with EF position as vector, you have to choose some coordinate system, otherwise some math won't make sense (mainly rotations)

> Is that because subsequent vector maths expect longitude first?

Not exactly. Vector math works in any combination, but direction of angles change (angle is from X to Y). 


> If so, we need to change the sequence in gpsSol.llh and gps_location_t, don't we?

Not necessarily. lat/lon are angles and we do not apply vector math to them. They are converted to some cartesian system first (is this PR GPS_distance2d converts to `EN(U)`) - you must always subtract lat/lon first = set reference point of conversion. lan/lon represent sphere, but locally, the sphere is flat enough to pretend it is flat/planar/2d

So we should be more careful when converting llh to ENU, but than it is simple XYZ

> Can't changes of this kind be done in a separate PR that provides consistency across the codebase?

Definitely. I'm trying to explain why continuing current practise of trying out signs and swapping sin/cos  until it works as desires is bad idea.
Poshold/rescue is revealing problems with current codebase. IMO we should start doing it 'right', or at least stop making situation even worse ...